hacknet-upgrade-manager.js

```diff
@@ -79,7 +79,7 @@ export function upgradeHacknet(ns, maxSpend, maxPayoffTimeSeconds = 3600 /* 3600
     for (var i = 0; i < ns.hacknet.numNodes(); i++) {
         let nodeStats = ns.hacknet.getNodeStats(i);
         if (formulas) { // When a hacknet server runs scripts, nodeStats.production lags behind what it should be for current ram usage. Get the "raw" rate
```

Comment:
Thanks for the PR! Indeed, it is a bug that if one has the formulas API, it dives into using the ns.formulas.hacknetServers.hashGainRate.

But actually, there's an even easier fix for this. We just need to add this check:
```suggestion
        if (formulas && ns.hacknet.hashCapacity() > 0) { // When a hacknet server runs scripts, nodeStats.production lags behind what it should be for current ram usage. Get the "raw" rate
```
That's because the nodeStats.production is already (corectly) set the correct money/sec gain rate for hacknet nodes. The correction below is *only* required for hacknet servers because when they are being used to run scripts, their hash rate temporarily drops, and we have to compute the *true* hash rate.